testkit/src/main/scala/org/bitcoins/testkit/core/gen/CreditingTxGen.scala

```diff
@@ -40,15 +48,34 @@ sealed abstract class CreditingTxGen {
     //note, cannot put a p2wpkh here
     Gen.oneOf(p2pkOutput,
               p2pkhOutput,
-              multiSigOutput, /*cltvOutput,*/ csvOutput)
+              multiSigOutput, /*cltvOutput,*/ csvOutput,
+              conditionalOutput)
   }
 
-  def nonP2SHOutput: Gen[BitcoinUTXOSpendingInfo] = {
-    Gen.oneOf(p2pkOutput,
-              p2pkhOutput,
-              multiSigOutput, /*cltvOutput,*/ csvOutput,
-              p2wpkhOutput,
-              p2wshOutput)
+  /** Only for use in constructing P2SH outputs */
+  private def nonP2SHOutput: Gen[BitcoinUTXOSpendingInfo] = {
+    Gen
+      .oneOf(p2pkOutput,
+             p2pkhOutput,
+             multiSigOutput, /*cltvOutput,*/ csvOutput,
+             conditionalOutput,
+             p2wpkhOutput,
+             p2wshOutput)
+      .suchThat(_.scriptPubKey.compactSizeUInt.toLong + CompactSizeUInt(
+        UInt64(520)).bytes.length < 520L)
+      .suchThat {
+        case P2SHNestedSegwitV0UTXOSpendingInfo(_,
+                                                _,
+                                                _,
+                                                _,
+                                                _,
+                                                _,
+                                                witness: P2WSHWitnessV0,
+                                                _) =>
+          witness.redeemScript.compactSizeUInt.toLong + CompactSizeUInt(
```

Comment:
I think the same thing here applies https://github.com/bitcoin-s/bitcoin-s/blob/d86acfffedda99ad73f6997fbf750f873f7ab2d3/core/src/main/scala/org/bitcoins/core/script/interpreter/ScriptInterpreter.scala#L352